src/ocean/meta/codegen/Identifier.d

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+
+    Utilities to get string representations of symbols.
+
+    Copyright:
+        Copyright (c) 2017 sociomantic Labs GmbH. All rights reserved.
+
+    License:
+        Boost Software License Version 1.0. See LICENSE_BOOST.txt for details.
+        Alternatively, this file may be distributed under the terms of the Tango
+        3-Clause BSD License (see LICENSE_BSD.txt for details).
+
+*******************************************************************************/
+
+module ocean.meta.codegen.Identifier;
+
+import ocean.meta.AliasSeq;
+import ocean.meta.codegen.CTFE;
+import ocean.meta.types.Qualifiers;
+import ocean.meta.traits.Function;
+
+version (UnitTest)
+{
+    import ocean.meta.types.Typedef;
+}
+
+/*******************************************************************************
+
+    Returns "name" (identifier) of a given symbol as string
+
+    When used with aggregate type field, will return only name of field itself,
+    without any qualification.
+
+    Params:
+        Sym = any symbol alias
+
+*******************************************************************************/
+
+public template identifier ( alias Sym )
+{
+    static if (is(typeof(Sym) == function))
+        const identifier = funcIdentifierHack!(Sym)();
+    else
+        const identifier = Sym.stringof;
+}
+
+///
+unittest
+{
+    class ClassName { int fieldName; }
+    void funcName ( ) { }
+    extern(C) void funcNameArgs ( int a, double b ) { }
```

Comment:
`static`? Not sure if it makes a difference here, nested `extern(C)` functions are unusual (if at all really possible in the sense that one can call them from an extern C program).